src/JWT/JwtDecoder.cs

```diff
@@ -178,9 +234,84 @@ public void Validate(JwtParts jwt, byte[] key)
 
             _jwtValidator.Validate(payloadJson, decodedCrypto, decodedSignature);
         }
+        
+        /// <summary>
+        /// Prepares data before calling <see cref="IJwtValidator.Validate" />.
+        /// </summary>
+        /// <param name="jwt">The JWT parts.</param>
+        /// <param name="keys">The keys provided which one of them was used to sign the JWT.</param>
+        /// <exception cref="ArgumentNullException" />
+        /// <exception cref="ArgumentOutOfRangeException" />
+        /// <exception cref="FormatException" />
+        public void Validate(JwtParts jwt, List<byte[]> keys)
+        {
+            if (jwt == null)
+                throw new ArgumentNullException(nameof(jwt));
+            if (keys == null || keys.Count == 0)
+                throw new ArgumentNullException(nameof(keys));
+            if(!DoesKeysHaveValues(keys))
+                throw new ArgumentOutOfRangeException(nameof(keys));
+
+            var crypto = _urlEncoder.Decode(jwt.Signature);
+            var decodedCrypto = Convert.ToBase64String(crypto);
+
+            var headerJson = GetString(_urlEncoder.Decode(jwt.Header));
+            var headerData = _jsonSerializer.Deserialize<Dictionary<string, object>>(headerJson);
+
+            var payload = jwt.Payload;
+            var payloadJson = GetString(_urlEncoder.Decode(payload));
+
+            var bytesToSign = GetBytes(String.Concat(jwt.Header, ".", payload));
+
+            var algName = (string)headerData["alg"];
+            var alg = _algFactory.Create(algName);
+
+            List<string> decodedSignatures = new List<string>();
+            foreach (byte[] key in keys)
+            {
+                var signatureData = alg.Sign(key, bytesToSign);
+                decodedSignatures.Add(Convert.ToBase64String(signatureData));

```

Comment:
Please use LINQ here

Reply:
Do you mean something like this `keys.ForEach(key=>decodedSignatures.Add(Convert.ToBase64String(alg.Sign(key, bytesToSign))));`
If yes; I won't be able to use `string[]`, I must use `List<string>`

Reply:
`keys.Select(key => alg.Sign(key, bytesToSign)).Select(sd => Convert.ToBase64String(sd)).ToArray()` should to the trick?

Reply:
In such case, is it ok to use a MethodGroup, so the end code will be:
`keys.Select(key => alg.Sign(key, bytesToSign)).Select(sd => Convert.ToBase64String(sd)).ToArray();`

Reply:
I frankly prefer do not use method groups but define lambda variable(s) explicitly, just for readability, specially by external users who might not be fully aware about the context.

Reply:
I get that, I reverted the code to remove the GroupMethods